src/main/java/emissary/util/FlexibleDateTimeParser.java

```diff
@@ -217,70 +194,80 @@ public static ZonedDateTime parse(final String dateString, final List<DateTimeFo
+     * +0000) and remove all text afterward. This should cover another set of cases where there is random text appended to
+     * the end of the string, as well as removing invalid non-numeric time zone offsets while still picking up the numeric
+     * offset Assumption - that tryExtensiveParsing is true - we should only get to this point if we want to try our best to
+     * parse
+     *
+     * @param date The date string to parse
+     * @return the ZonedDateTime object if removing text at the end was successful, or null otherwise
      */
-    private static void configure() {
-        try {
-            // fire up the configurator
-            Configurator configurator = ConfigUtil.getConfigInfo(FlexibleDateTimeParser.class);
-            setupTimezone(configurator.findStringEntry(CFG_TIMEZONE, DEFAULT_TIMEZONE));
-            setupDateFormats(configurator.findStringMatchEntries(CFG_FORMAT_MAIN), configurator.findStringMatchEntries(CFG_FORMAT_EXTRA));
+    @Nullable
+    static ZonedDateTime lastDitchParsingEffort(final String date) {
 
-            setupRemoveRegex(configurator);
-        } catch (IOException e) {
-            throw new IllegalArgumentException("Could not configure parser!!", e);
+        // Attempt to remove all text after the numeric offset and try again - this should give us a valid date string
+        // to work with
+        Matcher matcher = extraTextRemove.matcher(date);
+        if (matcher.find()) {
+            String secondChanceDate = matcher.replaceAll(matcher.group(1));
+            // if we removed text, attempt to parse again to see if we are more successful this time
+            return parsingHelper(secondChanceDate, true);
         }
+        return null;
     }
 
-    private static void setupRemoveRegex(Configurator configurator) {
-        String removeRegex = configurator.findStringEntry(CFG_REMOVE_REGEX, "<.+?>$|=0D$|\\(|\\)|\"|\\[|]|\\W+$|^\\W+");
-        remove = Pattern.compile(removeRegex, Pattern.DOTALL);
-        /*
-         * This is our last ditch parsing effort if we failed to parse the string - remove all extra text after the numeric time
-         * zone offset
-         */
-        String extraTextRemoveRegex = configurator.findStringEntry(CFG_EXTRA_TEXT_REMOVE_REGEX, "((\\+|-)\\d{4}).*$");
-        extraTextRemove = Pattern.compile(extraTextRemoveRegex);
+    /**
+     * Created to help against code duplication. Calls parse with the standard set of date formats, and then if that fails,
+     * attempt the extra set of date formats if tryExtensiveParsing is set to true.
+     *
+     * @param dateString The string we are attempting to parse
+     * @param tryExtensiveParsing Whether to use the extensive set of date formats
+     * @return The ZonedDateTime object if our parsing was successful, or null if not
+     */
+    private static ZonedDateTime parsingHelper(final String dateString, boolean tryExtensiveParsing) {
```

Comment:
Any chance we want to make a slightly-out-of-scope change and give this method a meaningful name such as `parseToZonedDateTime`?

Reply:
Done